pg_backup_api/pg_backup_api/server_operation.py

```diff
@@ -0,0 +1,163 @@
+
+import argparse
+import json
+import logging
+import os
+import sys
+import tempfile
+
+from datetime import datetime
+from os.path import join, exists
+
+from pg_backup_api.utils import (
+        barman,
+        load_barman_config,
+        get_server_by_name)
+load_barman_config()
+
+logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+log = logging.getLogger()
+
+
+DEFAULT_OP_TYPE = "recovery"
+JOBS_DIR = 'jobs'
+OUTPUT_DIR = 'output'
+REQUIRED_OPTIONS = ('backup_id', 'destination_directory', 'remote_ssh_command')
+
+
+class Metadata(object):
+    def __init__(self, server_name, operation_id=None):
+        self.operation_type = DEFAULT_OP_TYPE
+        self.operation_id = operation_id
+        self.server_name = server_name
+        self.server_config = get_server_by_name(server_name)
+        if not self.server_config:
+            raise ValueError("No barman config found for '{}'.".format(server_name))
+
+        BARMAN_HOME = barman.__config__.barman_home
+        self.jobs_basedir = join(BARMAN_HOME, server_name, JOBS_DIR)
+        self.output_basedir = join(BARMAN_HOME, server_name, OUTPUT_DIR)
+
+
+class ServerOperation(Metadata):
+    def get_operations_list(self):
+        jobs_list = []
+        if os.path.exists(self.jobs_basedir):
+            for job in [files for _, _, files in os.walk(self.jobs_basedir)][0]:
+                if job.endswith('.json'):
+                    operation_id, _ = job.split('.json')
+                    jobs_list.append(operation_id)
+        return jobs_list
+
+
+    def get_status_by_operation_id(self):
+        output_file = self.get_output_file()
+        if not os.path.exists(output_file):
+            if os.path.exists(self.get_job_file()):
+                return "IN_PROGRESS"
+            else:
+                raise Exception("Invalid operation-id")
+
+        with open(self.get_output_file()) as file_object:
+            content = json.load(file_object)
+            return content.get('success')
```

Comment:
That's what is going to happen because those values will be saved to the file, in the field named "success" so it whatever content we save there, is what the request will get once it queries.

But you're right I missed the "DONE", "FAILED" part.